modules/lang-expression/src/main/java/org/elasticsearch/script/expression/ExpressionSearchScript.java

```diff
@@ -63,15 +61,25 @@ public boolean needsScores() {
         return needsScores;
     }
 
+
     @Override
     public LeafSearchScript getLeafSearchScript(final LeafReaderContext leaf) throws IOException {
         return new LeafSearchScript() {
+            // Fake the scorer until setScorer is called.
+            DoubleValues values = source.getValues(leaf, new DoubleValues() {
+                @Override
+                public double doubleValue() throws IOException {
+                    return Double.NaN;
```

Comment:
Same here, we can't do that. The correct fix would be to pass a null value but it's also not possible currently because some DoubleValueSources that need to access the score check that the scorer is not null when `getValues` is called. We could also check if the source needs the score but this information is lost in the `ValueSource` to `DoubleValueSource` conversion because the converted `DoubleValueSource` always returns true when `needsScore` is invoked.